usaspending_api/etl/management/commands/load_multiple_submissions.py

```diff
@@ -88,33 +87,40 @@ def handle(self, *args, **options):
-            logger.info(
-                "Submission ID {} ({})\tCertified: {}".format(
-                    next_missing_sub[0], next_missing_sub[1], next_missing_sub[2].date()
+            except Exception:
+                logger.exception("Submission ID {} failed in pull from broker".format(submission_id))
+                failed_submissions.append(str(submission_id))
+
+        if len(missing_submissions):
+            logger.info("Total missing submissions: {}".format(len(missing_submissions)))
+            logger.info("-----------------------------------")
+            for submission in missing_submissions:
+                logger.info(
+                    "Submission ID {} ({})\tCertified: {}".format(submission[0], submission[1], submission[2].date())
                 )
-            )
-        logger.info("-----------------------------------")
+            logger.info("-----------------------------------")
+        else:
+            logger.info("No DABS to load")
 
         # Data modification happens here, if you don't flag '--safe'
-        # The submission loader is atomic, so one of these failing should not affect subsequent submissions
-        if not options["safe"]:
-            for next_missing_sub in missing_submissions:
-                submission_id = next_missing_sub[0]
-                try:
-                    call_command("load_submission", "--noclean", submission_id)
-                except Exception as error:
-                    logger.debug("Submission {} failed to load: {}".format(submission_id, error))
-                    failed_submissions.append(submission_id)
-
-        # If there were any failures, display them
-        if failed_submissions.__len__() > 0:
-            message = "The following submissions failed:\n"
-            message = message + reduce((lambda msg, failed_id: str(msg) + " " + str(failed_id)), failed_submissions)
-            logger.error(message)
-            exit(3)
+        # The submission loader is atomic, so one of these failing will not affect subsequent submissions
+        if options["safe"]:
+            logger.info("Exiting script before data load occurs in accordance with the --safe flag.")
+            return
+
+        for submission in missing_submissions:
+            submission_id = submission[0]
+            try:
+                call_command("load_submission", "--noclean", submission_id)
+            except SystemExit:
+                logger.info("Submission {} failed to load: {}".format(submission_id))
+                failed_submissions.append(str(submission_id))
+            except Exception:
+                logger.exception("Submission {} failed to load".format(submission_id))
+                failed_submissions.append(str(submission_id))
+
+        if failed_submissions:
+            logger.error("Script completed with the following submissions failures: {}".format(", ".join(failed_submissions)))
+            raise SystemExit(3)
```

Comment:
When run in a pipeline, will this make other tasks not run?

Reply:
I'm not following, this is the same functionality as previously.

Currently, the pipeline script is not coded to handle Status Code 3. It will be in the future.

Reply:
I misunderstood how exit() works in python. I see now that these are equivalent statements